packages/make-styles/src/runtime/resolveStaticStyleRules.ts

```diff
@@ -0,0 +1,28 @@
+import { expand } from 'inline-style-expand-shorthand';
+import { HASH_PREFIX } from '../constants';
+import { MakeStaticStyles, MakeStylesResolvedRule } from '../types';
+import { hashString } from './utils/hashString';
+import { compileStaticCSS } from './compileStaticCSS';
+
+export function resolveStaticStyleRules(
+  styles: MakeStaticStyles,
+  result: Record<string, MakeStylesResolvedRule> = {},
+): Record<string, MakeStylesResolvedRule> {
+  if (typeof styles === 'string') {
```

Comment:
_This will explode 💣_

---

Under hood we are using [`insertRule()`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule) in `.insertDefinitions`, the issue is that we cannot insert multiple rules with this API:

```js
$0.sheet.insertRule('body { background: red; }; div { color: red; }')
```

```
VM549:1 Uncaught DOMException: Failed to execute 'insertRule' on 'CSSStyleSheet': Failed to parse the rule 'body { background: red; }; div { color: red; }'.
    at <anonymous>:1:10
```

That's why Fela uses `.textContent` modification: https://github.com/robinweser/fela/blob/e5e96ab16e00f9dbe33f53faf2de475b2bd30023/packages/fela-dom/src/dom/connection/createSubscription.js#L44-L51

Another fancy option is to use Stylis (my darling ❤️ ): with `rulesheet()` plugin we can extract all rules one by one:
https://github.com/thysultan/stylis.js/blob/1f053d618e91e9da91295c9a50f2ce71a7d64c59/test/Middleware.js#L6-L15

Emotion uses this approach: https://github.com/emotion-js/emotion/blob/b59084b7cbfeab11c2a986033952d481d0b9fcf4/packages/cache/src/index.js#L139-L141

Reply:
great catch & thanks for the references! I updated the code to use `rulesheet ` plugin.